From 7b3586e1d6102d43f8a3683489570204d0bf1b43 Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:04 -0400
Subject: [fs] cifs: ntstatus_to_dos_map[] is not terminated

Message-id: <1406549822-10376-12-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86730
O-Subject: [RHEL7.1 PATCH 11/69] [bz1082049]cifs: ntstatus_to_dos_map[] is not terminated
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Tim Gardner <tim.gardner@canonical.com>

Functions that walk the ntstatus_to_dos_map[] array could
run off the end. For example, ntstatus_to_dos() loops
while ntstatus_to_dos_map[].ntstatus is not 0. Granted,
this is mostly theoretical, but could be used as a DOS attack
if the error code in the SMB header is bogus.

[Might consider adding to stable, as this patch is low risk - Steve]

Reviewed-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: Steve French <smfrench@gmail.com>
(cherry picked from commit 0c26606cbe4937f2228a27bb0c2cad19855be87a)

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/netmisc.c b/fs/cifs/netmisc.c
index af847e1..651a527 100644
--- a/fs/cifs/netmisc.c
+++ b/fs/cifs/netmisc.c
@@ -780,7 +780,9 @@ static const struct {
  ERRDOS, ERRnoaccess, 0xc0000290}, {
  ERRDOS, ERRbadfunc, 0xc000029c}, {
  ERRDOS, ERRsymlink, NT_STATUS_STOPPED_ON_SYMLINK}, {
- ERRDOS, ERRinvlevel, 0x007c0001}, };
+ ERRDOS, ERRinvlevel, 0x007c0001}, {
+ 0, 0, 0 }
+};
 
 /*****************************************************************************
  Print an error message from the status code
-- 
1.7.1